pkg/snapshot/snapshotter/snapshotter.go

```diff
@@ -736,19 +736,34 @@ func (ssr *Snapshotter) resetFullSnapshotTimer() error {
 	return nil
 }
 
+// checkSnapstoreSecretUpdate optimized by checking file modfication timestamp instead of computing the hash
 func (ssr *Snapshotter) checkSnapstoreSecretUpdate() bool {
-	ssr.logger.Debug("checking the hash of snapstore secret...")
-	newSnapstoreSecretHash, err := snapstore.GetSnapstoreSecretHash(ssr.snapstoreConfig)
+	ssr.logger.Debug("checking the timestamp of snapstore secret...")
+	newSnapstoreSecretModTime, err := snapstore.GetSnapstoreSecretModTime(ssr.snapstoreConfig)
 	if err != nil {
+		ssr.logger.Info("Error ", err.Error())
 		return true
```

Comment:
`true` is returned whenever `GetSnapstoreSecretModTime()` errors out because this will force a new `snapstore` object to be created before the delta snapshot is taken. This is useful since creation of a new `snapstore` object will:  
* ensure that all the dependent objects needed are healthy
* any entity which does not fit the expected format will cause errors

which will be propagated to the logs - thus reusing error handling and logging logic for the `snapstore` object instead of writing custom error handling for the case when credential files are ill formed when invoking `GetSnapstoreSecretModTime()`, with verbose logging.

Reply:
Marking this as resolved due to the above explanation.